core/rio/trig/src/main/java/org/eclipse/rdf4j/rio/trig/TriGWriter.java

```diff
@@ -100,6 +100,32 @@ public void handleStatement(Statement st)
 			throw new RuntimeException("Document writing has not yet been started");
 		}
 
+        // If we are pretty-printing, all writing is buffered until endRDF is
+        // called
```

Comment:
The trouble with that is that it isn't a full solution and will corrupt some datasets that rely heavily on blank node use if we anonymise any blank nodes at all. For example, there are parts of the OWL to RDF translation which insist on the use of blank nodes and not IRIs which could be corrupted by a buffering/anonymising solution  depending on how the statements are delivered. The pure streaming approach and the full buffering approach will never corrupt datasets that don't OOM.

There may need to be another option to switch on/off anonymising blank nodes that could be used to specify whether local grouping/buffering occurred rather than a full buffer.

One issue I have noticed is that Pretty-Printing setting is on by default, and we probably want to switch that in this pull request as I don't see pretty-printing as a production feature for any scalable system, even for the other formats.